lib/mysql/connector/network.py

```diff
@@ -220,10 +220,12 @@ def recv_plain(self):
         """Receive packets from the MySQL server"""
         try:
             # Read the header of the MySQL packet, 4 bytes
-            packet = bytearray(4)
-            read = self.sock.recv_into(packet, 4)
-            if read != 4:
-                raise errors.InterfaceError(errno=2013)
+            packet = bytearray()
+            while len(packet) < 4:
+                chunk = self.sock.recv(1)
+                if not chunk:
+                    raise errors.InterfaceError(errno=2013)
+                packet += chunk
```

Comment:
That makes sense ^^

On Fri, Feb 20, 2015, 19:55 INADA Naoki <notifications@github.com> wrote:

> In lib/mysql/connector/network.py
> <https://github.com/mysql/mysql-connector-python/pull/4#discussion_r25119048>
> :
>
> > @@ -220,10 +220,12 @@ def recv_plain(self):
> >          """Receive packets from the MySQL server"""
> >          try:
> >              # Read the header of the MySQL packet, 4 bytes
> > -            packet = bytearray(4)
> > -            read = self.sock.recv_into(packet, 4)
> > -            if read != 4:
> > -                raise errors.InterfaceError(errno=2013)
> > +            packet = bytearray()
> > +            while len(packet) < 4:
> > +                chunk = self.sock.recv(1)
> > +                if not chunk:
> > +                    raise errors.InterfaceError(errno=2013)
> > +                packet += chunk
>
> In typically case, sock.recv(4) returns 4 bytes.
> Calling socket.recv() 4 times may drop performance significantly.
>
> packet = b''while len(packet) < 4:
>     chunk = self.sock.recv(4-len(packet))
>     if not chunk:
>         raise errors.InterfaceError(errno=2013)
>     packet += chunk
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mysql/mysql-connector-python/pull/4/files#r25119048>.
>